shaders/src/post_process.vs

```diff
@@ -22,5 +22,13 @@ void main() {
     // Compute texel center
     vertex_uv = (floor(vertex_uv) + vec2(0.5, 0.5)) * frameUniforms.resolution.zw;
 #endif
+
     gl_Position = position;
+
+#if defined(TARGET_METAL_ENVIRONMENT)
+    // Metal texture space is vertically flipped that of OpenGL's, so flip the Y coords so we sample
+    // the frame correctly. Vulkan doesn't need this fix because its clip space is mirrored
+    // (the Y axis points down the screen).
+    gl_Position.y = -gl_Position.y;
```

Comment:
We'll have to be careful with this when multiple post-process steps are used. Also what happens when post-processing is completely disabled then? Or if FXAA is disabled but the rest of the post-processing is still enabled?

Reply:
The vertical flip compensates for Metal’s Y-inverted texture space, putting the UV origin of the fullscreen quad at the top left of the screen (as opposed to the bottom left for OpenGL). It makes no difference whether an even or odd number of post processing passes happen.

This is what the vertex shader for the fullscreen quad looks like:
![postprocess_vertex](https://user-images.githubusercontent.com/5298046/52587060-f0aed700-2ded-11e9-9a46-85c122a4620d.png)
Either way, the orientation is correct within each pass.

Reply:
Makes sense, thanks!

Reply:
BTW glslang can do this on our behalf, there's a special option for it.

Reply:
Yeah, the `flip_vert_y` flag? We only want to do it for the post-processing shaders so `GLSLPostProcessor` would need some logic to determine if it's a material shader or a post-processing one.